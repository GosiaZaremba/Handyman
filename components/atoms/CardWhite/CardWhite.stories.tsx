import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {CardWhite} from './CardWhite';

storiesOf('atoms/CardWhite', module)
  .addDecorator(story => story())
  .add('default', () => <CardWhite children={''} />);
