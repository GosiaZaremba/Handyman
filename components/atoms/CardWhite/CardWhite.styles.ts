import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.font.white,
    borderRadius: 5,
    width: '100%',
  },
});
