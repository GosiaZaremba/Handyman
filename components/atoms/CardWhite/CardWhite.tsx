import React from 'react';
import {View} from 'react-native';
import styles from './CardWhite.styles';

export type Props = {
  children: any;
};

export const CardWhite: React.FC<Props> = ({children}) => {
  return <View style={styles.container}>{children}</View>;
};
