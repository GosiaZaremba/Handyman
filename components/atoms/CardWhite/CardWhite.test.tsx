import React from 'react';
import {render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {CardWhite} from './CardWhite';
import {View} from 'react-native';

describe('Atom CardWhite', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <CardWhite testID="atom-cardWhite">
          <View />
        </CardWhite>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure children are rendered correctly', () => {
    const {getByTestId} = render(
      <CardWhite testID="atom-cardWhite">
        <View />
      </CardWhite>,
    );
    const cardWhite = getByTestId('atom-cardWhite');
    expect(cardWhite.props.children.type.displayName).toBe('View');
  });
});
