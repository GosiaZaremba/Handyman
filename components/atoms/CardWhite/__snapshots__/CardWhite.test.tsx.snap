// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom CardWhite Component Snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 15,
      "elevation": 8,
      "width": "100%",
    }
  }
  testID="atom-cardWhite"
>
  <View />
</View>
`;
